pyatlan/client/oauth.py

```diff
@@ -0,0 +1,90 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2025 Atlan Pte. Ltd.
+import threading
+import time
+from typing import Optional
+
+import httpx
+from authlib.oauth2.rfc6749 import OAuth2Token
+
+from pyatlan.client.constants import GET_OAUTH_CLIENT
+from pyatlan.utils import API
+
+
+class OAuthTokenManager:
+    def __init__(
+        self,
+        base_url: str,
+        client_id: str,
+        client_secret: str,
+        http_client: Optional[httpx.Client] = None,
+        connect_timeout: float = 30.0,
+        read_timeout: float = 900.0,
+    ):
+        self.base_url = base_url
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.token_url = self._create_path(GET_OAUTH_CLIENT)
+        self._lock = threading.Lock()
+        self._http_client = http_client or httpx.Client(
+            timeout=httpx.Timeout(
+                connect=connect_timeout, read=read_timeout, write=30.0, pool=30.0
```

Comment:
are these default values (`write=30.0, pool=30.0`) coming from `httpx` library or you're using arbitrary values here? I would prefer these to be configuration via `__init__()`:
```suggestion
                connect=connect_timeout, read=read_timeout, write=30.0, pool=30.0
```